### PR TITLE
Refactor/python docstrings for uov estimator

### PR DESCRIPTION
**Pull request by Dioprz**
### Description

Implementation of https://github.com/Crypto-TII/CryptographicEstimators/pull/169 for UOVEstimator.

### Review process

- A quick glance at the new docstrings would be great.
- Please search by FIX and TODO in the diff view, in case I has moved some previously existent comment.

**Estimator-specific doctests**
```shell
make docker-run
pytest --doctest-modules -n auto -vv cryptographic_estimators/UOVEstimator
```

**Cumulative test (with all the already migrated docstrings)**
```shell
make docker-pytest-doctests
```

### Pre-approval checklist
- [x] The code builds clean without any errors or warnings
- [x] I've added/updated tests
- [x] I've added/updated documentation if needed

### CODE REVIEW

**Review by Javierverbel — DISMISSED** [review]

**Review by Javierverbel — APPROVED** [review]

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py [hunk 1]
```diff
@@ -17,34 +17,28 @@
...
+        Collision attack is a general attack which works against any signature which 
+        follows the hash-and-sign paradigm. 
+
+        Args:
+            problem (UOVProblem): An instance of the UOVProblem class
+            **kwargs: Additional keyword arguments
+                gray_code_eval_cost (float): Logarithm of the cost to evaluate one polynomial in one vector using Gray-code enumeration (default: log(q))
+                X (int): Number of preimages
+                Y (int): Number of variables in the salt space
+                memory_access (int): Specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
```
Comment (FloydZ) [inline cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py:39]: wrong indentation

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py [hunk 2]
```diff
@@ -20,31 +20,27 @@
...
+
+        Args:
+            problem (UOVProblem): An instance of the UOVProblem class.
+            **kwargs: Additional keyword arguments.
+                w (int): Linear algebra constant (default: 2).
+                h (int): External hybridization parameter (default: 0).
+                excluded_algorithms (list): A list/tuple of MQ algorithms to be excluded (default: [Lokshtanov]).
+                memory_access (int): Specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage).
+                complexity_type (int): Complexity type to consider (0: estimate, 1: tilde O complexity, default: 0).
+                bit_complexities (int): Determines if complexity is given in bit operations or basic operations (default 1: in bit).
```
Comment (FloydZ) [inline cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py:42]: wrong indentation

File: cryptographic_estimators/UOVEstimator/uov_problem.py [hunk 3]
```diff
@@ -24,23 +24,20 @@
...
+            n (int): Number of variables
+            m (int): Number of polynomials
+            q (int): Order of the finite field
+            **kwargs: Additional keyword arguments
+                theta (float, optional): Exponent of the conversion factor. Defaults to 2.
+                    If 0 <= theta <= 2, every multiplication in GF(q) is counted as log2(q) ^ theta binary operation.
+                    If theta = None, every multiplication in GF(q) is counted as 2 * log2(q) ^ 2 + log2(q) binary operation.
+                cost_one_hash (int, optional): Bit complexity of computing one hash value. Defaults to 17.
+                memory_bound (float, optional): Maximum allowed memory to use for solving the problem. Defaults to inf.
+        """
```
Comment (FloydZ) [inline cryptographic_estimators/UOVEstimator/uov_problem.py:40]: wrong indentation, idk if this has any consequences on the docs

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/kipnis_shamir.py [hunk 4]
```diff
@@ -23,26 +23,23 @@
...
+        polynomial time. This key can then be used to generate signatures for arbitrary
+        messages.
+
+        Args:
+            problem (UOVProblem): An instance of the UOVProblem class.
+            **kwargs: Additional keyword arguments.
+                memory_access: Specifies the memory access cost model (default: 0, choices: 
+                    0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy 
+                    custom function which takes as input the logarithm of the total memory usage).
+                complexity_type: Complexity type to consider (0: estimate, 1: tilde O complexity, default: 0).
```
Comment (FloydZ) [inline cryptographic_estimators/UOVEstimator/UOVAlgorithms/kipnis_shamir.py:40]: wrong indentation

**Review by FloydZ — APPROVED** [review]

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/kipnis_shamir.py (hunk 4 above)
Comment (Javierverbel) [reply cryptographic_estimators/UOVEstimator/UOVAlgorithms/kipnis_shamir.py:40]: to be fixed in  #198

File: cryptographic_estimators/UOVEstimator/uov_problem.py (hunk 3 above)
Comment (Javierverbel) [reply cryptographic_estimators/UOVEstimator/uov_problem.py:40]: to be fixed in  #198

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py (hunk 2 above)
Comment (Javierverbel) [reply cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py:42]: to be fixed in  #198

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py (hunk 1 above)
Comment (Javierverbel) [reply cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py:39]: to be fixed in  #198

**Review by Javierverbel — APPROVED** [review]

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py (hunk 1 above)
Comment (Dioprz) [reply cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py:39]: Nothing to fix. Indentation is correctly parsed by our custom Sphinx extension (see https://github.com/Crypto-TII/CryptographicEstimators/pull/211).

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py (hunk 2 above)
Comment (Dioprz) [reply cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py:42]: Nothing to fix. Indentation is correctly parsed by our custom Sphinx extension (see https://github.com/Crypto-TII/CryptographicEstimators/pull/211).

File: cryptographic_estimators/UOVEstimator/uov_problem.py (hunk 3 above)
Comment (Dioprz) [reply cryptographic_estimators/UOVEstimator/uov_problem.py:40]: Nothing to fix. Indentation is correctly parsed by our custom Sphinx extension (see https://github.com/Crypto-TII/CryptographicEstimators/pull/211).

File: cryptographic_estimators/UOVEstimator/UOVAlgorithms/kipnis_shamir.py (hunk 4 above)
Comment (Dioprz) [reply cryptographic_estimators/UOVEstimator/UOVAlgorithms/kipnis_shamir.py:40]: Nothing to fix. Indentation is correctly parsed by our custom Sphinx extension (see https://github.com/Crypto-TII/CryptographicEstimators/pull/211).

**Pull request merged**